src/components/services/background/background.js

```diff
@@ -0,0 +1,104 @@
+angular.module("bawApp.components.background", [])
+    .service("bgImage", function() {
+        var curBg = "";
+        return {
+            curBg: curBg
+        };
+    })
+    .component("background",{
+        template: "<div ng-repeat='bg in bgs' ng-if=\"bg.state !== 'finished'\" class='background {{bg.state}}' style='background-image:url({{bg.url}});'></div>",
+        controller: ["$scope", "bgImage", function ($scope, bgImage) {
+
+            var self = this;
+
+            $scope.bgs = [];
+
+            /**
+             * Watches for changes to the bgImage service. When a change is detected it
+             * adds to the bgs array on scope. The array holds the url and the 'state'. State can be
+             * preloading, fadeIn, done and finished.
+             * preloading: image is preloading. Set the opacity to 0
+             * fadeIn: image has been preloaded, fade it in.
+             * done: image is finished fading in. Anything behind it can be removed.
+             * finished: is behind an image with state done. Remove it from dom.
+             * The bgs array is bound to scope in a repeat. New images can be added quicker than they can fade in without problem
+             * because it just adds them over the top of the existing fading-in images.
+             * @param newUrl
+             * @param oldUrl
+             */
+
+            self.setBg = function () {
+
+                if (typeof bgImage.curBg !== "string" || bgImage.curBg.length === 0) {
+                    return;
+                }
+
+                $scope.bgs.push({
+                    url : bgImage.curBg,
+                    state : "preloading"
+                });
+
+                var curBgNum = $scope.bgs.length - 1;
+
+                var preload = new Image();
+                preload.src = bgImage.curBg;
+
+                    if (preload.complete) {
+                        self.bgLoaded(curBgNum);
+                    } else {
+                        preload.addEventListener("load", function (e) {
+                            self.bgLoaded(curBgNum);
+                            this.remove();
```

Comment:
correct. It is the the image created with new Image() a few lines above. This is created only to preload the source so that it is in cache. I have read that removing prevents memory leaks.